File: EchoBot.cs
// EchoBot.cs
using System;
using System.Collections.Generic;

class EchoBot
{
    private List<string> history = new List<string>();
    private List<string> greetings = new List<string> { "Hello there! 👋", "Hi! 😊", "Greetings, human!", "Yo! 🤖", "Howdy!" };
    private Random rand = new Random();

    private void AddHistory(string msg)
    {
        history.Add(msg);
        if (history.Count > 20) history.RemoveAt(0);
    }

    public string Process(string text)
    {
        AddHistory(text);
        if (text.StartsWith("/"))
        {
            string[] parts = text.Split(' ', 2);
            string cmd = parts[0].ToLower();
            string arg = parts.Length > 1 ? parts[1] : "";
            switch (cmd)
            {
                case "/help": return Help();
                case "/greet": return greetings[rand.Next(greetings.Count)];
                case "/time": return DateTime.Now.ToString("yyyy-MM-dd HH:mm:ss");
                case "/history": return ShowHistory();
                case "/flip":
                    if (string.IsNullOrEmpty(arg)) return "Flip what? Type /flip <message>";
                    return SwapCase(arg);
                case "/bye": return "Goodbye! 👋";
                default: return $"Unknown command: {cmd}. Type /help for commands.";
            }
        }
        return $"You said: \"{text}\"";
    }

    private string Help()
    {
        return @"Available commands:
  /help      - Show this help
  /greet     - Get a random greeting
  /time      - Show current date/time
  /history   - Show recent messages
  /flip <msg> - Echo with swapped case
  /bye       - Exit the bot";
    }

    private string ShowHistory()
    {
        if (history.Count == 0) return "No messages yet.";
        List<string> lines = new List<string>();
        for (int i = 0; i < history.Count; i++)
            lines.Add($"  [{i+1}] {history[i]}");
        return "Recent messages:\n" + string.Join("\n", lines);
    }

    private string SwapCase(string s)
    {
        char[] arr = s.ToCharArray();
        for (int i = 0; i < arr.Length; i++)
        {
            if (char.IsLower(arr[i])) arr[i] = char.ToUpper(arr[i]);
            else if (char.IsUpper(arr[i])) arr[i] = char.ToLower(arr[i]);
        }
        return new string(arr);
    }

    static void Main()
    {
        var bot = new EchoBot();
        Console.WriteLine("🤖 Echo Bot: Hello! Type something or /help.");
        while (true)
        {
            Console.Write("You: ");
            string user = Console.ReadLine()?.Trim();
            if (string.IsNullOrEmpty(user)) continue;
            string response = bot.Process(user);
            if (user.Contains("/bye"))
            {
                Console.WriteLine($"🤖 Echo Bot: {response}");
                break;
            }
            Console.WriteLine($"🤖 Echo Bot: {response}");
        }
    }
}

File: echo_bot.py
# echo_bot.py
import datetime
import random
import sys

class EchoBot:
    def __init__(self):
        self.history = []
        self.greetings = ["Hello there! 👋", "Hi! 😊", "Greetings, human!", "Yo! 🤖", "Howdy!"]

    def _add_history(self, msg):
        self.history.append(msg)
        if len(self.history) > 20:
            self.history.pop(0)

    def process(self, text):
        self._add_history(text)
        if text.startswith("/"):
            parts = text.split(" ", 1)
            cmd = parts[0].lower()
            arg = parts[1] if len(parts) > 1 else ""
            if cmd == "/help":
                return self.help()
            elif cmd == "/greet":
                return random.choice(self.greetings)
            elif cmd == "/time":
                return datetime.datetime.now().strftime("%Y-%m-%d %H:%M:%S")
            elif cmd == "/history":
                return self.show_history()
            elif cmd == "/flip":
                return arg.swapcase() if arg else "Flip what? Type /flip <message>"
            elif cmd == "/bye":
                return "Goodbye! 👋"
            else:
                return f"Unknown command: {cmd}. Type /help for commands."
        else:
            return f"You said: \"{text}\""

    def help(self):
        return """Available commands:
  /help      - Show this help
  /greet     - Get a random greeting
  /time      - Show current date/time
  /history   - Show recent messages
  /flip <msg> - Echo with swapped case
  /bye       - Exit the bot"""

    def show_history(self):
        if not self.history:
            return "No messages yet."
        lines = [f"  [{i+1}] {msg}" for i, msg in enumerate(self.history)]
        return "Recent messages:\n" + "\n".join(lines)

def main():
    bot = EchoBot()
    print("🤖 Echo Bot: Hello! Type something or /help.")
    while True:
        try:
            user = input("You: ").strip()
            if not user:
                continue
            response = bot.process(user)
            if "/bye" in user:
                print(f"🤖 Echo Bot: {response}")
                break
            print(f"🤖 Echo Bot: {response}")
        except KeyboardInterrupt:
            print("\n🤖 Echo Bot: Goodbye! 👋")
            break
        except EOFError:
            break

if __name__ == "__main__":
    main()
